core/modules/wbase/Task.h

```diff
@@ -50,6 +50,28 @@ namespace lsst {
 namespace qserv {
 namespace wbase {
 
+/** Base class for tracking a database query for a worker Task.
+ */
+class TaskQueryRunner {
+public:
+    using Ptr = std::shared_ptr<TaskQueryRunner>;
+    virtual ~TaskQueryRunner() {};
+    virtual bool runQuery()=0;
+    virtual void cancel()=0;
+};
+
+class Task;
+
+/** Base class for scheduling Tasks.
+ * Allows the scheduler to take appropriate action when a task is cancelled.
+ */
+class TaskScheduler {
```

Comment:
again could be just a std::function I think if you buy into that.  Otherwise, you've used struct for some of these elsewhere, so if we're keeping them we should use struct or class consistently?